File: Data Structures & Algorithms/network-delay-time/submission-2.cpp
class Solution {
public:
    int networkDelayTime(vector<vector<int>>& times, int n, int k) {
        priority_queue<int> q;
        vector<int> dist(n+1,9999);
        dist[k] = 0;
        for(int i=1;i<=n;i++){

            for(int j=0;j<times.size();j++){
                int a = times[j][0], b= times[j][1], c= times[j][2];

                if(dist[b]>dist[a]+c){
                    dist[b] = dist[a] + c;
                }
            }

        }
        
        int ans = 0;
        for(int i=1;i<n+1;i++){
            ans = max(ans,dist[i]);
        }

        return ans==9999?-1:ans;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s)<=1:
            return len(s)
        st = 0
        en = 1
        ans = 1
        mp = {s[st]:1}
        while en<len(s):
            dv = mp.get(s[en],0)
            if dv!=1:
                ans = max(ans,en-st+1)
                mp[s[en]] = 1
            else:
                while st<en and mp[s[en]]==1:
                    cr = s[st]
                    mp[cr] = 0
                    st+=1
                mp[s[en]] = 1
            en+=1
        return ans


        